src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js

```diff
@@ -397,7 +399,21 @@ class ActivityFeed extends React.Component<Props, State> {
                             onAnnotationStatusChange={onAnnotationStatusChange}
                             onAppActivityDelete={onAppActivityDelete}
                             onCommentDelete={hasCommentPermission ? onCommentDelete : noop}
-                            onCommentEdit={hasCommentPermission ? onCommentUpdate : noop}
+                            onCommentEdit={
+                                hasCommentPermission && onCommentUpdate
```

Comment:
This is doing a lot. Can we treat this like the other props e.g. onCommentDelete, onAnnotationEdit and just create a new object argument type for onCommentEdit to ensure the right params are passed in.

Perhaps something like this added to types.js:
```

export type DiscussionItemType = {
    hasMention?: boolean,
    id: string,
    onError?: Function,
    onSuccess?: Function,
    permissions: BoxCommentPermission | AnnotationPermission,
    status?: FeedItemStatus,
    text?: string,
};
export type OnAnnotationEdit = DiscussionItemType => void;

export type OnCommentEdit = DiscussionItemType => void;

```

Reply:
Hmm, yeah, I don't like this part of the PR either, but I felt like replacing the args with a single arg all the way up was too much of a change, especially because it would result in a mismatch between the comment-related function and other functions at the respective levels... if that makes sense. That said, I'm happy to do it. But I want to make sure we're on the same page: 

I have already created an `OnCommentEdit` type[ in types.js](https://github.com/box/box-ui-elements/pull/3331/files#diff-8006f38d16ee725c1654fe66120517d595055479ef90050057a83fb337c1dafa), although it is important to note that it _cannot_ be the same type as `OnAnnotationEdit`. So, in the area that we are dealing with, `onCommentEdit` does take a single arg with props. But I had to pick a place to transform it back to args (which is what I'm doing here) unless we wanted to change the comment-related function to take an arg with props everywhere (and then it would be kind of out of place alongside the other functions in those files that still take args). 

I'm super tired. Kid kept me up. So I hope that makes sense. Just lmk which way you'd like me to go :)